src/reducers/documents.js

```diff
@@ -0,0 +1,133 @@
+import { createReducer } from "redux-action"
+
+export const CREATE_DOCUMENT = "CREATE_DOCUMENT"
+export const REMOVE_DOCUMENT = "REMOVE_DOCUMENT"
+export const SELECT_DOCUMENT = "SELECT_DOCUMENT"
+export const UPDATE_DOCUMENT = "UPDATE_DOCUMENT"
+export const CHANGE_TO_ADD_MODE = "CHANGE_TO_ADD_MODE"
+export const TITLE_INPUT_CHANGE = "TITLE_INPUT_CHANGE"
+export const TEXT_INPUT_CHANGE = "TEXT_INPUT_CHANGE"
+
+const initialState = {
+  list: [],
+  selected: {},
+  mode: 'edit',
+  titleInput: '',
+  textInput: '',
+}
+
+export const createDocument = (_document) => {
+  return {
+    type: CREATE_DOCUMENT,
+    document: _document,
+  }
+}
+
+export const updateDocument = (_document) => {
+  return {
+    type: UPDATE_DOCUMENT,
+    document: _document,
+  }
+}
+
+export const removeDocument = (documentId) => {
+  return {
+    type: REMOVE_DOCUMENT,
+    documentId,
+  }
+}
+
+export const selectDocument = (_document) => {
+  return {
+    type: SELECT_DOCUMENT,
+    document: _document,
+  }
+}
+
+export const changeToAddMode = () => {
+  return {
+    type: CHANGE_TO_ADD_MODE,
+  }
+}
+
+export const changeTitleInput = (value) => {
+  return {
+    type: TITLE_INPUT_CHANGE,
+    value
+  }
+}
+
+export const changeTextInput = (value) => {
+  return {
+    type: TEXT_INPUT_CHANGE,
+    value
+  }
+}
+
+let newList
+
+export default createReducer(initialState, ({
+  [CREATE_DOCUMENT]: (payload, state) => {
+    newList = state.list
```

Comment:
don't mutate the state - you can use spread operator instead:

newList = [...state.list]